flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -93,7 +93,12 @@
 	 * It can be specified using Scala's FiniteDuration notation: "1 min", "20 s".
 	 */
 	@PublicEvolving
-	public static final String RESTART_STRATEGY_FAILURE_RATE_DELAY = "restart-strategy.failure-rate.delay";
```

Comment:
For converting constants to options so far we followed a different approach: add new ConfigOptions into the appropriate `*Options` class, deprecate the existing constant in `ConfigConstants` and link to the new option.
Ultimately, the goal is for ConfigConstants to be fully deprecated _at some point_.

This may be a good opportunity to move all restart-related options into a dedicated or existing *Options class. This would have the added benefit that they'd be covered by the docs generator.

Reply:
I think this deserves to be a separate issue because it also requires updating the failover strategy documentation where we currently list the configuration values. The thing I want to improve is simply setting the default value to `0s`.

Reply:
I agree that changing the type is not a nice thing and I mainly followed #5031. But in order to do it properly I guess you are right that we need to create `RestartStrategyOptions`. I'll create a JIRA issue for it and base this PR on it.